CMakeLists.txt

```diff
@@ -269,6 +269,11 @@ set(BUILD_SHARED_LIBS true)
 set(OPENMS_HOST_DIRECTORY "${PROJECT_SOURCE_DIR}")
 set(OPENMS_HOST_BINARY_DIRECTORY "${PROJECT_BINARY_DIR}")
 
+# The additional / is important to remove the last character from the path.
+# Note that it does not matter if the OS uses / or \, because we are only
+# saving the path size.
+string(LENGTH "${PROJECT_SOURCE_DIR}/" CF_SOURCE_PATH_SIZE)
```

Comment:
dangerous.
In some (future) cases, a source file might also be configured and compiled from the build-tree.
If that build tree is in a shorter directory, then you might cut away parts of the filename or worse: point behind the last `\0` and funny things will happen.
So we'd need a check that the build-tree path length is longer, and if its not .. well... then advance only as much as its safe (minimum of both paths). This will look a bit weird in the final result, but better than undefined behaviour.
Maybe issue a warning in the CMake code if this happens..

Reply:
I can also add another macro for the build tree, if you want.

Reply:
I also found some solution with some constexpr magic but most of them assume "/" as the only separator.

Reply:
> 
> 
> I can also add another macro for the build tree, if you want.

Then the user would need to know when to use the source tree vs the build tree macro, right? Not ideal, and just as dangerous?

Reply:
ok, great fix. thanks!
i've added a line of docs